testing_guide.md

```diff
@@ -0,0 +1,698 @@
+
+_For feedback on this guide, please visit [#26092](https://github.com/bitcoin/bitcoin/issues/26092)_
+
+This document outlines some of the upcoming Bitcoin Core 24.0 release changes and provides steps to help test them. This guide is meant to be the starting point for experimentation and further testing, but is in no way comprehensive! After running through the steps in this guide, you are encouraged to do your own testing.
+
+This can be as simple as testing the same features in this guide but trying it a different way. Even better, think of features you use regularly and test that they still work as expected in the release candidate. You can also read the [release notes][release notes] to find something not covered in this guide. **This is a great way to be involved with Bitcoin's development and helps keep Bitcoin running smoothly and bug-free! Your help in this endeavor is greatly appreciated.**
+
+## Overview
+
+Changes covered in this testing guide include:
+
+- Observing the new headers pre-synchronization phase during IBD ([#25717](https://github.com/bitcoin/bitcoin/pull/25717))
+- Using the GUI to restore a wallet from a backup file ([#471](https://github.com/bitcoin-core/gui/pull/471))
+- Peristent settings are now unified between bitcoind and the GUI. ([#15936](https://github.com/bitcoin/bitcoin/pull/15936),[#602](https://github.com/bitcoin-core/gui/pull/602))
+- Testing transient addresses for I2P outbound connections ([#25355](https://github.com/bitcoin/bitcoin/pull/25355))
+- Using the new `migratewallet` RPC to migrate legacy wallets to descriptor wallets. ([#19602](https://github.com/bitcoin/bitcoin/pull/19602))
+- Testing watch-only support for Miniscript descriptors. ([#24148](https://github.com/bitcoin/bitcoin/pull/24148))
+
+For a comprehensive list of changes in Bitcoin Core 24.0, check out the [release notes][release notes].
+
+## Preparation
+
+### 1. Grab Latest Release Candidate
+
+**Current Release Candidate:** [Bitcoin Core 24.0rc1][source-code] ([release-notes][release notes])
+
+There are two ways to grab the latest release candidate: pre-compiled binary or source code. The source code for the latest release can be grabbed from here: [latest release source code][source-code]
+
+If you want to use a binary, make sure to [grab the correct one for your system][binaries].
+
+[source-code]: https://github.com/bitcoin/bitcoin/releases/tag/v24.0rc1
+[binaries]: https://bitcoincore.org/bin/bitcoin-core-24.0/test.rc1/
+
+### 2. Compile Release Candidate
+
+**If you grabbed a binary, skip this step**
+
+Before compiling, make sure that your system has all the right dependencies installed. As this guide utilizes the Bitcoin Core GUI, you must **compile with support for the GUI**(`--with-gui=yes`) and have the `qt5` dependency already installed. Since descriptor wallets are now the default, it is recommended that you **compile with sqlite**(`--with-sqlite=yes`), so make sure you have installed the `sqlite3` dependency.
```

Comment:
The issue would be on many linux distros where the bdb version we want isn’t available. You could install a new version and pass ‘with-incompatible-bdb’ to configure. Alternatively, there’s a helper script to grab and compile the bdb we want.

but there’s an easy solution to any user configuration issues, point them to our depends build system :D

Reply:
omg, I was clicking the link in the "Releases" column (which takes me to the Oracle site) instead of the link in the  "Dependency" column (which takes me to the helper script) in the [dependencies](https://github.com/bitcoin/bitcoin/blob/master/doc/dependencies.md) doc, and I somehow missed it on the installation doc for my distro. Totally agree on just pointing to the depends build system 👍

Reply:
Mentioning BDB for legacy wallets is indeed useful, I've just added that.